src/app/broadcaster.service.spec.ts

```diff
@@ -23,10 +22,13 @@ describe('Service: Broadcaster service', () => {
     broadcaster.broadcast('testEvent', 1001);
   });
 
-  it('Broadcaster can only be created once', () => {
-    expect(() => {
-      new Broadcaster();
-    }).toThrow();
+  it('Broadcaster can only be created once', (done) => {
+    broadcaster.on('cheese').subscribe((data: string) => {
+      expect(data).toEqual('cheddar');
+      done();
+    });
+
+    new Broadcaster().broadcast('cheese', 'cheddar');
```

Comment:
you need to add a counter in the callback and expect it to be 1 after calling broadcast (to check it has actually been called)

Reply:
my bad I didn't see `done()`

Reply:
calling the constructor here for a second time and seeing that the listener is still called validates that there is only 1 instance.